### ISSUE TITLE
Focus on speeding up WooCommerce

### ISSUE THREAD

**Issue by paaljoachim** [woothemes/woocommerce#48773]
### Describe the solution you'd like

I would like a speedier backend for all products screen as well as individual products screens.

I have had two clients complain about the slowness of WooCommerce products screens and individual screens.
It would be great for WooCommerce to look into how to speed up the backend (perhaps also frontend).

### Describe alternatives you've considered

Searching online for tips and tricks on how to speed up a slow backend WooCommerce all Product pages and individual Product pages.

### Additional context

WP 6.5.4
WooCommerce 9.01
None default theme

**Comment by naman03malhotra** [comment]
Hey @paaljoachim, thanks for reaching out to us. Can you please be more specific about the problem you are facing, it can include details like, slow queries and slow network calls. 

You can use [this plugin](https://wordpress.org/plugins/query-monitor/), to gather such info.

You can also use light speed in the bowser to collect client-side performance data.

**Comment by paaljoachim** [comment]
Hi @naman03malhotra 

I have installed Query Monitor. On the Installed plugins screen I see the following.

Queries by Component.
![Screenshot 2024-06-29 at 14 33 14](https://github.com/woocommerce/woocommerce/assets/5323259/deb8d5fa-082c-4743-89f0-b40a27a6eda2)

The above time went up to 1.4657 when on the All Products screen. Showing 30 items per screen.

Duplicate Queries
![Screenshot 2024-06-29 at 14 47 29](https://github.com/woocommerce/woocommerce/assets/5323259/5ffaccec-eecb-41e3-b68a-729b021a65f5)

------

Using Inspect -> Network I see the following slow load of the All Products page. I sorted the rows by time it took to load.
![Network console sorted by time all products screen](https://github.com/woocommerce/woocommerce/assets/5323259/4942c05d-47b3-4b9f-bc23-ca58952c4408)

Load time: 7.74

**Comment by paaljoachim** [comment]
Now going to an individual product that uses the theme Enfold which also adds some extra styling to it one gets this load time.

![Edit product Enfold load time](https://github.com/woocommerce/woocommerce/assets/5323259/5551f92a-616c-4680-b4e5-25b8fa54d384)

Load: 19.41 s

Let me know what else is needed. I can also give access to the test site.

**Comment by paaljoachim** [comment]
Now adding in this plugin: https://github.com/lukecav/performance-improvements-for-woocommerce/
Then going to All Products. Opening the Dev -> Network I see this:
![Adding in Lukes plugin load time goes down](https://github.com/woocommerce/woocommerce/assets/5323259/da821359-8fd4-41c5-9db5-c97017c26f66)

Load: 5.74s

An obvious difference in load times. Going from 7.74 to 5.74.

----------

Testing the individual Product page which uses Enfold. With Luke's plugin active.
![Screenshot 2024-06-29 at 15 39 10](https://github.com/woocommerce/woocommerce/assets/5323259/51d16cb2-b52b-4aac-938c-a9778eaf1693)

The load time goes down dramatically when Luke's plugin is active.
From Load: 19.41 s all the way down to 8.57s.

I also tested Luke's plugin on another site.
With Luke's plugin: Load times where somewhere between 11 - 15 seconds. 
Without Luke's plugin: Load times where around 17 seconds.

Conclusion is that Luke has been able to improve the load time of All products and individual product pages.
https://github.com/lukecav/performance-improvements-for-woocommerce/

I have also come across this plugin:
https://wordpress.org/plugins/disable-dashboard-for-woocommerce/

I see that WooCommerce has some work to do to improve the load time of the various backend WooCommerce pages.

**Comment by amplify11dev** [comment]
+1 for this, I flagged the network requests doubling on _every_ request even ones that aren't woo related https://github.com/woocommerce/woocommerce/issues/45729 and not a ton of movement even though it's a pretty significant and obvious bug. Many clients complaining about speed of Woo lately and I've seen it myself. If you run a flamegraph there's tons of stuff like loading the woo blocks that take half a second. There's also references to things like maybe loading PayPal even on sites without any PayPal enabled.

**Comment by mamedul1** [comment]
The performance bottlenecks you're seeing in the WooCommerce backend, particularly with high query counts on the product screens, are frequently caused by bloated `wp_options` and unindexed `postmeta` data. Since you've already identified duplicate queries via Query Monitor, the next step is to optimize the database structure and clean up autoloaded data that slows down the admin dashboard.

You can find a detailed breakdown of how to handle these specific database optimizations and general scaling strategies here:
- https://needlecode.com/blog/wordpress/database-optimization-woocommerce-wp-options-postmeta.html
- - https://needlecode.com/blog/woocommerce/optimizing-woocommerce-high-traffic-scale.html

**Comment by pmcpinto** [comment]
Thanks for the report, it's really helpful. I know this has been open for a long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

I'd like to get an engineer's eyes on this before we make a call, so I've flagged it for developer feedback. Someone will take a proper look at the repro steps and how it's behaving now.

**Comment by vladolaru** [comment]
Thanks for the detailed follow-up and screenshots. I took another pass through this and I think this issue is best treated as a broad performance umbrella rather than one directly actionable implementation ticket.

There are a few different surfaces mixed together here:

1. `Products > All Products` list-table request/query performance.
2. Individual classic product edit screen load/render performance, including the Enfold-theme example.
3. Broader WooCommerce Admin / REST request overhead, which was also mentioned in the thread via #45729.

For the **All Products** screen, several relevant performance changes have landed since the original report:

- #65276 added product-list cache priming and reported reducing the admin products page from `159` to `137` SQL queries in its test data.
- #65957 introduced persistent product status counters to avoid the degrading product status count query on large catalogs.
- #66176 disabled category-filter counters on the admin products page, removing another catalog-size-dependent query.

The corresponding code is on current trunk:

- Products list cache warm-up hooks: [class-wc-admin-list-table-products.php#L63-L65](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/list-tables/class-wc-admin-list-table-products.php#L63-L65)
- Product status count cache usage: [ProductUtil.php#L141-L154](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Utilities/ProductUtil.php#L141-L154)
- Product count cache lifecycle hooks: [ProductCountCacheService.php#L55-L68](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Caches/ProductCountCacheService.php#L55-L68)
- Category filter counters disabled: [class-wc-admin-list-table-products.php#L432-L445](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/list-tables/class-wc-admin-list-table-products.php#L432-L445)

For the **REST/admin request overhead** part, #45729 was partly addressed by #60684, which lazy-loads WooCommerce Analytics REST API namespaces. The broader "too many wc-admin / wc-analytics requests" concern is still better tracked in #60393 than in this product-screen umbrella.

For the **individual product edit screen**, I did not find a current focused fix that clearly addresses the Enfold product-edit load case from this report. That part needs a fresh, focused profile before it can be routed well. Useful data would be:

- current WooCommerce version / trunk retest;
- default theme result versus Enfold result;
- product count, variation count, attribute count, and taxonomy count;
- Query Monitor slow queries and duplicate queries;
- browser Performance profile or network waterfall for the product edit screen;
- active plugins and which parts of the comparison plugin materially change the profile.

I would not use the comparison plugin as a direct patch plan because it disables many independent WooCommerce/Admin features at once. It is useful evidence that some WooCommerce-admin work is contributing to the slowdown, but each change needs to be isolated before we can tell which one belongs in Core.

My recommendation is:

- treat the All Products part as likely materially improved by #65276, #65957, and #66176, then retest on current trunk if it is still slow;
- route REST/admin request-count concerns to #60393;
- split the individual product edit screen case into a focused issue with current profiling data if it still reproduces.
